### ISSUE TITLE
Infinite Loop

### ISSUE THREAD

**Issue by maximilianschmitt** [eush77/regexp.execall#1]
Hi! This code seems to cause an infinite loop:

``` js
'use strict';
var execAll = require('regexp.execall');
console.log(execAll(/.*/ig, 'i am the real slim shady'));
```

Best,
Max

**Comment by maximilianschmitt** [comment]
Actually, I think I might be using it wrong. I'm sorry, I have the same problem using this code:

``` js
var match;
while ((match = /.*/ig.exec(input)) !== null) {
  console.log(match);
}
```

**Comment by eush77** [comment]
``` js
> re = /.*/ig
/.*/gi
> re.exec('i am the real slim shady')
[ 'i am the real slim shady',
  index: 0,
  input: 'i am the real slim shady' ]
> re.exec('i am the real slim shady')
[ '', index: 24, input: 'i am the real slim shady' ]
> re.exec('i am the real slim shady')
[ '', index: 24, input: 'i am the real slim shady' ]
> re.exec('i am the real slim shady')
[ '', index: 24, input: 'i am the real slim shady' ]
```

Interesting. I think it makes sense to end matching if matches start to repeat themselves. This is a property of nullable regexps I wasn't familiar with before.

**Comment by maximilianschmitt** [comment]
Yeah, I'm unsure what causes this but it would be great if it didn't happen. :)

**Comment by eush77** [comment]
Fixed in 1.0.1. Thanks!

**Upd**: didn't quite get it right the first time. I think it's gone now in 1.0.2.

**Comment by maximilianschmitt** [comment]
Man, that is some great support, thanks a lot for implementing this so quickly! :)
